modules/Microsoft.HealthDataServices/deploy.bicep

```diff
@@ -0,0 +1,58 @@
+@description('Required. The name of the Health Data Services Workspace service.')
+@maxLength(50)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@allowed([
+  ''
+  'CanNotDelete'
+  'ReadOnly'
+])
+@description('Optional. Specify the type of lock.')
+param lock string = ''
+
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+@description('Optional. Control permission for data plane traffic coming from public networks while private endpoint is enabled.')
+param publicNetworkAccess string = 'Disabled'
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+// =========== //
+// Deployments //
+// =========== //
+
```

Comment:
Please add the `resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {` resource in the same way you did for the children also in this top-level module. 

However, note, that we don't want to bloat the deployments with the PID, so for any module that has children, we also have a variable `var enableReferencedModulesTelemetry = false` that we'd pass into any child-module to make sure the PID deployment only happens once, and not multiple times. As a reference you can see it e.g. [here](https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.KeyVault/vaults/deploy.bicep#:~:text=secret.roleAssignments%20%3A%20%5B%5D-,enableDefaultTelemetry%3A%20enableReferencedModulesTelemetry,-%7D). Happy to elaborate further it that sounds too cryptic.